backend/app/main.py

```diff
@@ -27,5 +28,26 @@ async def lifespan(app: FastAPI):
 
 app = FastAPI(lifespan=lifespan)
 
+
+@app.middleware("http")
+async def add_request_id(request: Request, call_next):
+    request_id = str(uuid.uuid4())
+    
+    logger.info(f"Request ID: {request_id} - {request.method} {request.url.path}")
+    
+    start_time = time.time()
+    response = await call_next(request)
+    process_time = time.time() - start_time
+    formatted_process_time = f"{process_time:.4f}s"
+    
+    logger.info(
+        f"Request ID: {request_id} - Completed {request.method} {request.url.path} "
+        f"Status Code: {response.status_code} Process Time: {formatted_process_time}"
+    )
+    
+    response.headers["X-Request-ID"] = request_id
+    return response
+
```

Comment:
This is exactly what we'll want for tracing and observability, and you've captured most of the fields I can think of that we'll need